co-simulation/lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/transform/Proj4Projection.java

```diff
@@ -0,0 +1,195 @@
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+ package org.eclipse.mosaic.lib.transform;
+
+import static java.lang.Math.toDegrees;
+import static java.lang.Math.toRadians;
+
+
+import org.eclipse.mosaic.lib.geo.CartesianPoint;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.geo.MutableCartesianPoint;
+import org.eclipse.mosaic.lib.geo.MutableGeoPoint;
+import org.eclipse.mosaic.lib.geo.MutableUtmPoint;
+import org.eclipse.mosaic.lib.geo.UtmPoint;
+import org.eclipse.mosaic.lib.geo.UtmZone;
+import org.eclipse.mosaic.lib.math.MathUtils;
+import org.eclipse.mosaic.lib.math.Vector3d;
+
+
+import org.locationtech.proj4j.CRSFactory;
+import org.locationtech.proj4j.CoordinateReferenceSystem;
+import org.locationtech.proj4j.CoordinateTransform;
+import org.locationtech.proj4j.CoordinateTransformFactory;
+import org.locationtech.proj4j.ProjCoordinate;
+
+
+
+public class Proj4Projection extends GeoProjection {
+
+    // private final String georeference;
+    private GeoPoint geoOrigin;
+    UtmZone zone;
+
+    private double x_offset;
+    private double y_offset;
+
+    private String wgs84ReferenceString;
+    private String cartesianReferenceString;
+    CoordinateReferenceSystem wgs84CRS;
+    CoordinateReferenceSystem cartesianCRS;
+
+
+    public Proj4Projection(GeoPoint origin, double x_offset, double y_offset, String georef){
+        this.geoOrigin = origin;
+        this.x_offset = x_offset;
+        this.y_offset = y_offset;
+
+        // The proj library being used currently doesn't support use of vertical keywords
+        //such as geodgrids and vunits, resulting in no vertical accuracy in the transforms calculated
+        //Isssue tracked under: https://github.com/locationtech/proj4j/issues/20
+        //Remove unaccepted parameters from georeference
+        String regex = "(\\+geoidgrids=[^\\s]+\\s?)|(\\+vunits=[^\\s]+\\s?)";
+
+        this.cartesianReferenceString = georef.replaceAll(regex, "").trim();;
+        this.wgs84ReferenceString = "EPSG:4326";
+
+        CRSFactory crsFactory = new CRSFactory();
+        this.cartesianCRS = crsFactory.createFromParameters("custom_proj", this.cartesianReferenceString);
+        this.wgs84CRS = crsFactory.createFromName(wgs84ReferenceString);
+
+    }
+
+    @Override
+    public Vector3d geographicToVector(GeoPoint geographic, Vector3d result){
+        getGeoCalculator().distanceBetween(geoOrigin, geographic, result);
+        return result;
+    }
+
+    @Override
+    public MutableGeoPoint vectorToGeographic(Vector3d vector3d, MutableGeoPoint result) {
+        getGeoCalculator().pointFromDirection(geoOrigin, vector3d, result);
```

Comment:
Where is this geocalculator defined? What transform is it using, or is it not using a transform at all?

Reply:
This defaults to using a default geocalculator which makes calculation using the reference ellipsoid https://github.com/usdot-fhwa-stol/cdasim/blob/aa079b58017d71b6e8341cf133750efa8ca06408/co-simulation/lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/transform/GeoProjection.java#L63

Updated the implementation now to use the projections defined by proj (initialized in the constructor)